### PR TITLE
Validate Lab input : allow only one decimal value

### PR DESCRIPTION
**Pull request by Shreets**
image
image

### CODE REVIEW

File: src/pages/LabIntakeInput.js [hunk 1]
```diff
@@ -1,13 +1,145 @@
...
+  neutrophilError,
+  setNeutrophilError,
+  lymphocytesError,
+  setLymphocytesError,
+  eosinophilsError,
+  setEosinophilsError,
+  basophilsError,
+  setBasophilsError,
+  plateletsError,
+  setPlateletsError,
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:31]:
```suggestion
  labErrors,
  setLabErrors,
```

File: src/pages/LabIntakeInput.js [hunk 2]
```diff
@@ -1,13 +1,145 @@
...
+
+  const handleEsrChange = (e) => {
+    const item = e.target.name;
+    if (e.target.value === "" || e.target.value.match(DECIMAL_REGEX)) {
+      setEsrError(false);
+    } else {
+      setEsrError(true);
+    }
+    setLabState({ ...labState, [item]: e.target.value });
+  };
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:51]: please create one function and use it for all inputs:
```suggestion
  const handleEsrChange = (e) => {
    const item = e.target.name;
    const value = e.target.value;
   
    setLabErrors(errors => ({ ...errors, [item]: value === "" || value?.match(DECIMAL_REGEX)?.[0] }));
    setLabState(state => ({ ...state, [item]: value }));
  };
```

File: src/pages/LabIntakeInput.js [hunk 3]
```diff
@@ -21,9 +153,14 @@ const LabIntakeInput = ({ labState, setLabState }) => {
                   type="number"
                   inputMode="decimal"
                   name="crp"
-                  onChange={handleInputChange}
+                  onChange={handleCrpChange}
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:156]:
```suggestion
                  onChange={handleLabInputChange}
```

File: src/pages/LabIntakeInput.js [hunk 4]
```diff
@@ -21,9 +153,14 @@ const LabIntakeInput = ({ labState, setLabState }) => {
...
                   value={labState["crp"]}
                 />
+                {crpError ? (
+                  <p className="lab-error-message">
+                    upto one decimal point only
+                  </p>
+                ) : null}
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:163]: Please update all inputs similarly:
```suggestion
                {labErrors.crp && (
                  <p className="lab-error-message">
                    upto one decimal point only
                  </p>
                )}
```

**Review by nastikue — CHANGES_REQUESTED** [review]
@Shreets please get rid of duplicating code

File: src/pages/LabIntakeInput.js (hunk 1 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

File: src/pages/LabIntakeInput.js (hunk 2 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

File: src/pages/LabIntakeInput.js (hunk 3 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

**Comment by Shreets** [comment]
> @Shreets please get rid of duplicating code

The duplicates have been removed

File: src/pages/LabIntakeInput.js [hunk 5]
```diff
@@ -1,13 +1,24 @@
...
-  const handleInputChange = (e) => {
+const LabIntakeInput = ({ labState, setLabState, labError, setLabError }) => {
+  const handleLabInputChange = (e) => {
     const item = e.target.name;
+    const value = e.target.value;
+
+    setLabError((errors) => ({
+      ...errors,
+      [item]:
+        e.target.value === "" || e.target.value.match(DECIMAL_REGEX)
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:15]:
```suggestion
        value === "" || value.match(DECIMAL_REGEX)
```

File: src/pages/LabIntakeInput.js [hunk 6]
```diff
@@ -1,13 +1,24 @@
...
+          ? false
+          : true,
+    }));
     setLabState({ ...labState, [item]: e.target.value });
```
Comment (nastikue) [inline src/pages/LabIntakeInput.js:19]:
```suggestion
    setLabState({ ...labState, [item]: value });
```

File: src/pages/LabIntakeInput.js (hunk 5 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

File: src/pages/LabIntakeInput.js (hunk 6 above)
Comment (Shreets) [reply src/pages/LabIntakeInput.js]: resolved

**Review by nastikue — APPROVED** [review]

**Pull request merged**